workflow-runtime/src/commonMain/kotlin/com/squareup/workflow1/internal/WorkflowNode.kt

```diff
@@ -226,15 +227,26 @@ internal class WorkflowNode<PropsT, StateT, OutputT, RenderingT>(
   }
 
   /**
-   * Applies [action] to this workflow's [state] and
-   * [emits an output to its parent][emitOutputToParent] if necessary.
+   * Applies [action] to this workflow's [state] and then passes the resulting [ActionApplied]
+   * via [emitAppliedActionToParent] to the parent, with additional information as to whether or
+   * not this action has changed the current node's state.
    */
   private fun applyAction(
-    action: WorkflowAction<PropsT, StateT, OutputT>
-  ): ActionProcessingResult? {
-    val (newState, outputOrNull) = action.applyTo(lastProps, state)
+    action: WorkflowAction<PropsT, StateT, OutputT>,
+    childResult: ActionApplied<*>? = null
+  ): ActionProcessingResult {
+    val (newState, actionApplied) = action.applyTo(lastProps, state)
```

Comment:
```suggestion
    val (newState: StateT, actionApplied: ActionApplied<StateT>) = action.applyTo(lastProps, state)
```

Not positive about the `ActionApplied` type, but you get the idea.